src/GitVersionCore/VersionCalculation/IncrementStrategyFinder.cs

```diff
@@ -131,5 +131,16 @@ private static IEnumerable<ICommit> GetIntermediateCommits(IGitRepository repo,
             if (none.IsMatch(message)) return VersionField.None;
             return null;
         }
+
+        private static IEnumerable<ICommit> GetCommitsReacheableFromHead(IGitRepository repo, ICommit headCommit)
```

Comment:
This method can be moved to `IGitRepository.Commits.ReachableFromHead()`.

Reply:
I think I will keep it like this in the PR, so that we can have more time to think.